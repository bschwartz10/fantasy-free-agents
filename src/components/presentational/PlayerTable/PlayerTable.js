import React from 'react';
import { Table } from 'react-bootstrap';
import Proptypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import PlayerRow from '../PlayerRow';

const PlayerTable = ({ players, handleClick }) => {
  return (
    <Table responsive>
      <thead>
        <tr>
          <th className="name">Name <FontAwesomeIcon onClick={(e) => handleClick(e, 'nameSorted', 'name')} icon="sort" /></th>
          <th className="position">Position <FontAwesomeIcon onClick={(e) => handleClick(e, 'positionSorted', 'position')} icon="sort" /></th>
          <th className="team">Team <FontAwesomeIcon onClick={(e) => handleClick(e, 'teamSorted', 'team')} icon="sort" /></th>
          <th className="height">Height <FontAwesomeIcon onClick={(e) => handleClick(e, 'heightSorted', 'height')} icon="sort" /></th>
          <th className="weight">Weight <FontAwesomeIcon onClick={(e) => handleClick(e, 'weightSorted', 'weight')} icon="sort" /></th>
          <th className="adp">Adp <FontAwesomeIcon onClick={(e) => handleClick(e, 'adpSorted', 'adp')} icon="sort" /></th>
        </tr>
      </thead>
      <tbody>
        {players.map(player => <PlayerRow player={player} key={player.id} />)}
      </tbody>
    </Table>
  );
}

PlayerTable.propTypes = {
  players: Proptypes.array,
}

export default PlayerTable;
