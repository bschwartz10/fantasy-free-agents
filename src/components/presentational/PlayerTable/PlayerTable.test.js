import React from 'react'
import { shallow, mount } from 'enzyme'
import PlayerTable from './PlayerTable'

describe('PlayerTable component', () => {
  let wrapper;
  const players = [
    {
      adp: "400",
      birthdate: "656226000",
      cbs_id: "1977109",
      college: "UCLA",
      draft_team: "FA",
      draft_year: "2012",
      espn_id: "15351",
      height: "72",
      id: "11524",
      jersey: "48",
      kffl_id: "26333",
      name: "Derrick Coleman",
      nfl_id: "derrickcoleman/2534871",
      position: "RB",
      rotoworld_id: "8019",
      sportsdata_id: "e0a63251-5428-43a1-88c1-c000215ac5ce",
      stats_global_id: "461705",
      stats_id: "26299",
      team: "ARI",
      weight: "233"
    },
    {
      adp: "1.38",
      birthdate: "775890000",
      cbs_id: "2000877",
      college: "Georgia",
      draft_team: "STL",
      draft_year: "2015",
      espn_id: "2977644",
      height: "73",
      id: "12150",
      jersey: "30",
      kffl_id: "32621",
      name: "Todd Gurley",
      position: "RB",
      rotoworld_id: "10288",
      sportsdata_id: "14263792-d1d3-4b0c-85f7-2a85b4aed6f1",
      stats_global_id: "694641",
      stats_id: "28398",
      team: "LAR",
      weight: "227",
    }
  ]

  beforeEach(() => {
    wrapper = shallow(<PlayerTable players={players} handleClick={jest.fn()} />)
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  it('render a row for each player', () => {
    expect(wrapper.find('tbody').children().length).toEqual(players.length)
  })

  it('render table with correct headers', () => {
    expect(wrapper.find('.name').children().first().text().trim()).toEqual('Name')
    expect(wrapper.find('.position').children().first().text().trim()).toEqual('Position')
    expect(wrapper.find('.team').children().first().text().trim()).toEqual('Team')
    expect(wrapper.find('.height').children().first().text().trim()).toEqual('Height')
    expect(wrapper.find('.weight').children().first().text().trim()).toEqual('Weight')
    expect(wrapper.find('.adp').children().first().text().trim()).toEqual('Adp')
  })


  it('should match snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

})
