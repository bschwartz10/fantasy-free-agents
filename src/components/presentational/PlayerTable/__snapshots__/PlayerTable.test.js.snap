// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerTable component should match snapshot with all data passed in correctly 1`] = `
<Table
  bordered={false}
  bsClass="table"
  condensed={false}
  hover={false}
  responsive={true}
  striped={false}
>
  <thead>
    <tr>
      <th
        className="name"
      >
        Name 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="sort"
          inverse={false}
          listItem={false}
          mask={null}
          onClick={[Function]}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </th>
      <th
        className="position"
      >
        Position 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="sort"
          inverse={false}
          listItem={false}
          mask={null}
          onClick={[Function]}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </th>
      <th
        className="team"
      >
        Team 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="sort"
          inverse={false}
          listItem={false}
          mask={null}
          onClick={[Function]}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </th>
      <th
        className="height"
      >
        Height 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="sort"
          inverse={false}
          listItem={false}
          mask={null}
          onClick={[Function]}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </th>
      <th
        className="weight"
      >
        Weight 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="sort"
          inverse={false}
          listItem={false}
          mask={null}
          onClick={[Function]}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </th>
      <th
        className="adp"
      >
        Adp 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="sort"
          inverse={false}
          listItem={false}
          mask={null}
          onClick={[Function]}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
        />
      </th>
    </tr>
  </thead>
  <tbody>
    <PlayerRow
      key="11524"
      player={
        Object {
          "adp": "400",
          "birthdate": "656226000",
          "cbs_id": "1977109",
          "college": "UCLA",
          "draft_team": "FA",
          "draft_year": "2012",
          "espn_id": "15351",
          "height": "72",
          "id": "11524",
          "jersey": "48",
          "kffl_id": "26333",
          "name": "Derrick Coleman",
          "nfl_id": "derrickcoleman/2534871",
          "position": "RB",
          "rotoworld_id": "8019",
          "sportsdata_id": "e0a63251-5428-43a1-88c1-c000215ac5ce",
          "stats_global_id": "461705",
          "stats_id": "26299",
          "team": "ARI",
          "weight": "233",
        }
      }
    />
    <PlayerRow
      key="12150"
      player={
        Object {
          "adp": "1.38",
          "birthdate": "775890000",
          "cbs_id": "2000877",
          "college": "Georgia",
          "draft_team": "STL",
          "draft_year": "2015",
          "espn_id": "2977644",
          "height": "73",
          "id": "12150",
          "jersey": "30",
          "kffl_id": "32621",
          "name": "Todd Gurley",
          "position": "RB",
          "rotoworld_id": "10288",
          "sportsdata_id": "14263792-d1d3-4b0c-85f7-2a85b4aed6f1",
          "stats_global_id": "694641",
          "stats_id": "28398",
          "team": "LAR",
          "weight": "227",
        }
      }
    />
  </tbody>
</Table>
`;
